### ISSUE TITLE
cleaning job archive does not reflect split-user

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#936]
Originally reported of Google Code with ID 880.

```
What software version are you running? Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

# gnt-cluster --version
gnt-cluster (ganeti v2.9.6) 2.9.6

# gnt-cluster version
Software version: 2.9.6
Internode protocol: 2090000
Configuration format: 2090000
OS api version: 20
Export interface: 0
VCS version: v2.9.6

# hspace --version
hspace (ganeti) version v2.9.6
compiled with ghc 7.6
running on linux x86_64

<b>What distribution are you using?</b>
# cat /etc/*release
SUSE Linux Enterprise Server 11 (x86_64)
VERSION = 11
PATCHLEVEL = 3

<b>What steps will reproduce the problem?</b>
1. build ganeti with a split-user setup (--with-user-prefix= --with-group-prefix=)
2. install sample doc/examples/ganeti.cron
3. watch logfile /var/log/ganeti/master-cleaner/cleaner-*.log

<b>What is the expected output? What do you see instead?</b>

The resulting cron-tab runs master-cleaner as ${user-prefix}-masterd:

45 1 * * * gnt-masterd [ -x /usr/sbin/ganeti-cleaner ] && /usr/sbin/ganeti-cleaner master

The log /var/log/ganeti/master-cleaner/cleaner-2014-07-14T11_09.23674.log says:
Cleaner started at Mon Jul 14 11:09:01 CEST 2014
find: cannot stat current directory: Permission denied
find: cannot stat current directory: Permission denied

For any reason find does open(".", O_RDONLY) and newfstatat(AT_FDCWD, ".", 0x627a60, AT_SYMLINK_NOFOLLOW). Under cron CWD is /var/spool/cron and this is not accessable by the ${user-prefix}-masterd user.

One soloution cloud be to change ganeti-cleaner, so that $CWD is something like $DATA_DIR.

Thanks, Sascha.
```
Originally added on 2014-07-14 10:33:38 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-07-15 09:30:54 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Can you please try if the following patch fixes the issue?
```
Originally added on 2014-07-15 09:57:14 +0000 UTC.

Attachments: 
[0001-Make-ganeti-cleaner-switch-to-a-save-working-directo.patch](https://storage.googleapis.com/google-code-attachments/ganeti/issue-880/comment-2/0001-Make-ganeti-cleaner-switch-to-a-save-working-directo.patch)

**Comment by Ganeti-Issues-Migrator** [comment]
```
Patch workes as intended.

Thanks, Sascha.
```
Originally added on 2014-07-15 12:10:14 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit faa5ce00af6dfa3f729a845b0a6526c7a2cabaaf
Author: Klaus Aehlig <aehlig@google.com>
Date:   Tue Jul 15 11:40:52 2014 +0200

    Make ganeti-cleaner switch to a save working directory
    
    For some reason, find does stat the current working directory,
    which, when invoked via cron, may or may not be readable by the
    process (issue 880). Therefore, first change to a directory that
    is definitely accessible to the process; the directory it logs
    into is such a safe place.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2014-07-16 10:29:53 +0000 UTC.

Mentioned Issue with GoogleCode ID 880 has been migrated to GitHub with ID #936.
Changed State: Fixed
Added to Milestone: Release2.9

**Comment by Ganeti-Issues-Migrator** [comment]
```
released as part of 2.11.6
```
Originally added on 2014-09-22 15:58:21 +0000 UTC.

Changed State: Released
